src/capability.rs

```diff
@@ -83,22 +73,40 @@ impl NativeCapability {
         instance: impl CapabilityProvider,
         binding_name: Option<String>,
     ) -> Result<Self> {
-        let capid = instance.capability_id();
+        let b: Box<dyn CapabilityProvider> = Box::new(instance);
+        let descriptor = get_descriptor(&b)?;
+        let binding = binding_name.unwrap_or("default".to_string());
+
+        info!(
+            "Loaded native capability provider '{}' v{} ({}) for {}/{}",
+            descriptor.name, descriptor.version, descriptor.revision, descriptor.id, binding
+        );
         Ok(NativeCapability {
-            capid: capid.to_string(),
-            plugin: Box::new(instance),
-            binding_name: binding_name.unwrap_or("default".to_string()),
+            descriptor,
+            plugin: b,
+            binding_name: binding,
             library: None,
         })
     }
 
     /// Returns the capability ID (namespace) of the provider
     pub fn id(&self) -> String {
-        self.capid.to_string()
+        self.descriptor.id.to_string()
     }
 
     /// Returns the human-friendly name of the provider
     pub fn name(&self) -> String {
-        self.plugin.name().to_string()
+        self.descriptor.name.to_string()
     }
+
+    /// Returns the full descriptor for the capability provider
+    pub fn descriptor(&self) -> &CapabilityDescriptor {
+        &self.descriptor
+    }
+}
+
+fn get_descriptor(plugin: &Box<dyn CapabilityProvider>) -> Result<CapabilityDescriptor> {
+    let res = plugin.handle_call("system", OP_GET_CAPABILITY_DESCRIPTOR, &[])?;
```

Comment:
As I recall there was talk at one time of making `system` a constant defined somewhere.
Would now be a time to do that?

Reply:
Added https://github.com/wascc/wascc-codec/issues/13 to capture this.